src/server/apsystem.ts

```diff
@@ -343,7 +344,7 @@ export default class ActivityPubSystem {
     if (moderationState === BLOCKED) {
     // TODO: Notify of blocks?
       await this.rejectActivity(fromActor, activityId)
-    } else if (moderationState === ALLOWED) {
+    } else if ((activityType === 'Undo') || (moderationState === ALLOWED)) {
```

Comment:
so it's undone because it was previously approved, and there's no point in rejecting it since we were rejecting activities anyway? i think they should always be approved if they pass some checks (the undo and the undone activities are from the same actor) regardless of current actor moderation status